src/Command/DebugCommand.php

```diff
@@ -28,7 +28,36 @@ public function __construct(ContainerInterface $container, array $listeners, arr
                 $listener = $container->get($serviceId);
                 $method = array_key_exists('method', $tag) ? $tag['method'] : '__invoke';
 
-                $events[$tag['event']][] = get_class($listener) . '::' . $method;
+                if (isset($tag['event'])) {
+                    $events[$tag['event']][] = get_class($listener) . '::' . $method;                    
+                }
+
+                if (isset($tag['typehints'])) {
+                    $reflClass = new ReflectionClass($container->getParameterBag()->resolveValue($listener));
+
+                    foreach ($reflClass->getMethods() as $method) {
+                        if (!$method->isPublic()
+                            || $method->isConstructor()
+                            || $method->isStatic()
+                            || $method->isAbstract()
+                            || $method->isVariadic()
+                            || $method->getNumberOfParameters() !== 1
```

Comment:
Should we handle methods like [`__get()`](https://www.php.net/manual/ru/language.oop5.overloading.php#object.get) here?

Reply:
I think I would not know how to do it. I did a tribute (copy + paste + adapt) to the typehint function that uses the tactician-bundle  https://github.com/thephpleague/tactician-bundle/blob/bf10a37e27e4306bb9de5d909dba36226aa8e3c4/src/DependencyInjection/HandlerMapping/TypeHintMapping.php